### PR TITLE
Upgrade to modern javascript

### LINKED ISSUE #417: bootstrap-editable is not compatible with bootstrap 4+
https://vitalets.github.io/x-editable/

Possibly replace with custom stimulus

### LINKED ISSUE #426: nested_form gem is obsolete
It has been archived https://github.com/ryanb/nested_form

### PR DESCRIPTION
**Pull request by jcoyne**
Using stimulus + turbo frames to replace old functionality.
Fixes #417 
Fixes #426

### CODE REVIEW

File: app/javascript/controllers/hello_controller.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Controller } from "@hotwired/stimulus"
```
Comment (marlo-longley) [inline app/javascript/controllers/hello_controller.js:1]: Is this a boilerplate file?

**Comment by marlo-longley** [comment]
@jcoyne I am still looking at the code, but for me these red and green "Remove location" and "Add location" buttons on the "Edit library" page do not work. For remove, it does nothing. For add, it submits something that reloads/redirects to the parent page and I get the "success" alert, but I wasn't actually given the form for adding a library. 


Screenshot 2023-01-19 at 16 23 15

**Comment by jcoyne** [comment]
the "create library" text is what shows in prod now, right?

**Comment by marlo-longley** [comment]
@jcoyne yes -- feel free to ignore. Could be a separate small PR.

**Review by marlo-longley — APPROVED** [review]

**Pull request merged**